### PR TITLE
update source reference functionality

### LINKED ISSUE #1294: Fix query intelligence for new `FROM` query syntax in DuckDB v0.6
To give us more time to test it, we decided to merge DuckDB v0.6 to main before fixing a few cosmetic bugs related to its new `FROM` statement (see: https://github.com/rilldata/rill-developer/pull/1254). They are:

- [ ] Sources are not displayed when we don't use SELECT because of the SQL parser
- [ ] No information about change in row counts is displayed when when we don't use SELECT because of the SQL parser

### PR DESCRIPTION
**Pull request by hamilton**
closes #1294

the update to duckdb `0.6.0` causes issues with our ad-hoc "extract table names from query" function. This PR:
1. replaces the older code with a much simpler regular expression
2. updates the frontend to consume this new regular expression to match tables & pair with existing sources

### CODE REVIEW

File: web-local/src/lib/util/get-table-references/index.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ *
+ * @param sql the sql query
+ * @returns {Reference[]} an array of reference locations
+ */
+export function getTableReferences(sql: string) {
+  if (!sql) return [];
+  // eslint-disable-next-line no-useless-escape
+  const regex = /(?:from|join)\s+([a-zA-z0-9_.]+|"[a-zA-z0-9\.\-_\/:\s]+")/gim;
```
Comment (AdityaHegde) [inline web-local/src/lib/util/get-table-references/index.ts:16]: Would this be able to extract the tables from a CTE? Would be good to have those tests from the previous suite.

File: web-local/src/lib/components/Editor.svelte [hunk 2]
```diff
@@ -309,9 +309,9 @@
 
   function underlineSelection(selections: any) {
```
Comment (AdityaHegde) [inline web-local/src/lib/components/Editor.svelte:310]: Better to type to get a sense of what `selections` is and understand the underlining code at a glance.

File: web-local/src/lib/util/get-table-references/index.ts (hunk 1 above)
Comment (hamilton) [reply web-local/src/lib/util/get-table-references/index.ts:16]: yes, it should work in the case of CTEs as well. There's actually nothing special about CTEs with this approach. I can add those tests though for the sake of clarity

File: web-local/src/lib/components/Editor.svelte (hunk 2 above)
Comment (hamilton) [reply web-local/src/lib/components/Editor.svelte:315]: Attempted to add the `SelectionRange` type and uncovered more issues. I've added a FIXME for now. This is not ideal but we'll need to go a bit further up the chain to resolve the type issues here.

File: web-local/src/lib/util/get-table-references/index.ts (hunk 1 above)
Comment (hamilton) [reply web-local/src/lib/util/get-table-references/index.ts:16]: I've added a CTE test to the suite.

**Comment by hamilton** [comment]
@AdityaHegde give me a minute, going to fix the remaining features that required the table references function.

**Review by AdityaHegde — APPROVED** [review]

File: web-local/src/lib/util/get-table-references/test-data.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      referenceIndex: 17,
+    },
+  ],
+};
+
+export const fromInCTE: ReferenceTestCase = {
+  query: `WITH x as (select * from tbl) FROM x`,
+  references: [
+    { reference: "tbl", type: "from", index: 20, referenceIndex: 25 },
+    { reference: "x", type: "from", index: 30, referenceIndex: 35 },
```
Comment (AdityaHegde) [inline web-local/src/lib/util/get-table-references/test-data.ts:125]: Found the issue after merging, with the new approach returning CTE aliases we would end up marking the model dependant on any table that might have the same name as the alias.

**Pull request merged**